hepstats/hypotests/fitutils/sampling.py

```diff
@@ -0,0 +1,62 @@
+from .api_check import is_valid_pdf
+
+"""
+Module providing basic sampling methods.
+"""
+
+
+def base_sampler(models, nevents, floating_params=None):
+    """
+    Creates a samplers from models.
+
+    Args:
+        models (list): models to sample
+        nevents (list): number of in each sampler
+        floating_params (list): floating parameter in the samplers
+    """
+
+    assert all(is_valid_pdf(m) for m in models)
+    assert len(nevents) == len(models)
+
+    if floating_params:
+        floating_params = [f.name for f in floating_params]
```

Comment:
This means that the list of floating parameters is a list of specific objects. Be explicit :-).